master/internal/db/postgres_trial.go

```diff
@@ -533,9 +393,30 @@ func (db *PgDB) _addTrialMetricsTx(
 				"rollback updating latest validation ID for trial %d: %w", m.TrialId, err)
 		}
 
-		if err := db.fullTrialSummaryMetricsRecompute(ctx, tx, int(m.TrialId)); err != nil {
-			return rollbacks, errors.Wrap(err, "error on rollback compute of summary metrics")
+		trainSummary, err := db.fullTrialSummaryMetricsRecompute(ctx, tx, int(m.TrialId),
```

Comment:
suggestion: IMO this (parent) function is already doing a lot and I'd lean toward not adding more to it. what do you think about keeping the full recompute logic out of there like before?